File: 26/26_remove_duplicates.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0;
        for (int j=1; j<nums.size(); j++) {
            if (nums[j] != nums[i]) {
                i++;
                nums[i] = nums[j];
            }
        }
        return i + 1;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {0, 0, 1, 1, 1, 1, 2, 3, 4};
    int size = sol.removeDuplicates(nums);
    cout << size << endl;
}

File: 26/26_remove_duplicates.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 1
        for j in range(1, len(nums)):
            if nums[j] != nums[j-1]:
                nums[i] = nums[j]
                i+=1
        return i



if __name__ == "__main__":
    sol = Solution()
    nums = [0, 0, 1, 1, 1, 1, 2, 3, 4, 4]
    print(sol.removeDuplicates(nums))